stdlib/public/core/StringWordView.swift

```diff
@@ -0,0 +1,202 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension String {
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public struct _WordView {
```

Comment:
Hm, it may be worth considering putting this through Swift Evolution and making it fully public in this release. I know we have too many proposals in flight already, but it feels bad to use up an ABI stable bit in `String.Index` for a feature that isn't public. (E.g., we don't need that bit unless there is a risk someone will actually feed invalid indices to the word view. Since we are in full control over SPI use sites, we could also choose to rather require them to explicitly call a special conversion/rounding method instead.)

Reply:
If we can't get it through SE, is there any reason this core functionality can't be a few SPI functions rather than a whole type?

Reply:
Yep, we can also just have methods for advancing an index to the next/previous word boundary